lib/upload.js

```diff
@@ -242,6 +272,16 @@ class Upload {
       clearTimeout(this._retryTimeout);
       this._retryTimeout = null;
     }
+
+    if (shouldTerminate) {
+      Upload.terminate(this.url, this.options, (err, xhr) => {
+        if (err || !this._hasStorage()) {
+          return cb(err, xhr);
+        }
+
+        this._storage.removeItem(this._fingerprint, cb);
+      });
+    }
```

Comment:
The callback should also be invoked if `shouldTerminate === false`, just to have a consistent API interface.

Reply:
fixed